src/shared/bus-locator.c

```diff
@@ -81,6 +88,31 @@ const BusLocator* const bus_hostname = &(BusLocator){
         .interface = "org.freedesktop.hostname1"
 };
 
+BusLocator *bus_locator_new(const char *dest, const char *iface, const char *path) {
```

Comment:
I think this is not useful. Why not just store the bus path in struct? There's no need to make everything into a struct with destructors/...

Reply:
I can't speak as to why this was added originally (I've just rebased this), but I agree it's simpler to use the bus path and have removed the constructor/destructor commit for `BusLocator`.